dts/dts.cmake

```diff
@@ -16,25 +16,23 @@ message(STATUS "Generating zephyr/include/generated/generated_dts_board.h")
 
 if(CONFIG_HAS_DTS)
 
-  set(DTC_INCLUDE_FLAG_FOR_SOURCE
-    -include ${DTS_SOURCE}
-    )
-
-  if(NOT DTC_OVERLAY_FILE)
-    # overlay file is not set, so try to use overlay dir instead.
-
-    set_ifndef(DTC_OVERLAY_DIR ${APPLICATION_SOURCE_DIR})
-
-    set(DTC_OVERLAY_FILE ${DTC_OVERLAY_DIR}/${BOARD_FAMILY}.overlay)
+  set(BOARD_OVERLAY_FILE ${APPLICATION_SOURCE_DIR}/${BOARD_FAMILY}.overlay)
+  if(NOT EXISTS ${BOARD_OVERLAY_FILE})
+    unset(BOARD_OVERLAY_FILE)
   endif()
 
-  if(EXISTS ${DTC_OVERLAY_FILE})
-    set(DTC_INCLUDE_FLAG_FOR_OVERLAY
-      -include ${DTC_OVERLAY_FILE}
-      )
-  else()
-    unset(DTC_INCLUDE_FLAG_FOR_OVERLAY)
-  endif()
+  set(
+    dts_files
+    ${DTS_SOURCE}
+    ${BOARD_OVERLAY_FILE}
+    ${DTC_OVERLAY_FILE}
+  )
+
+  unset(DTC_INCLUDE_FLAG_FOR_DTS)
+  foreach(dts_file ${dts_files})
```

Comment:
Assuming you want DTC_OVERLAY_FILE to behave like CONF_FILE (i.e. use whitespace instead of semicolons to separate), then it needs some post-processing to make a DTC_OVERLAY_AS_LIST or so. Here's a pointer:

https://github.com/zephyrproject-rtos/zephyr/blob/master/cmake/kconfig.cmake#L13

Reply:
Yep, I accidentally removed that actually, I had it locally. Fixed